fe/fe-core/src/main/java/com/starrocks/metric/MaterializedViewMetricsEntity.java

```diff
@@ -282,17 +280,13 @@ public Integer getValue() {
                 }
                 MaterializedView mv = (MaterializedView) table;
 
-                Locker locker = new Locker();
-                locker.lockDatabase(db, LockType.READ);
                 try {
                     if (!mv.getPartitionInfo().isPartitioned()) {
```

Comment:
why remove it ? what if update the partitioninfo concurrently ?

Reply:
if mv partition info is changed, can we just create a new one from the old one, then remove the old one after quiescent period ?

Reply:
fixed.